src/main/java/engineering/everest/starterkit/filestorage/config/DeduplicatingFileStoreConfig.java

```diff
@@ -16,23 +16,21 @@
 public class DeduplicatingFileStoreConfig {
 
     @Bean
-    @Qualifier("permanentDeduplicatingFileStore")
-    DeduplicatingFileStore permanentFileStore(FileMappingRepository fileMappingRepository,
-                                              @Qualifier("permanentFileStore") FileStore fileStore) {
+    DefaultDeduplicatingFileStore permanentFileStore(FileMappingRepository fileMappingRepository,
+                                                     @Qualifier("permanentFileStore") FileStore fileStore) {
         return new DefaultDeduplicatingFileStore(PERMANENT, fileMappingRepository, fileStore);
```

Comment:
We no longer need to pass the filestore type in as a parameter :smile:

Reply:
The filestore here is for pointing to appropriate beans which are picked conditionally from MongoGridFsFileStoreConfig or AwsS3FileStoreConfig. So, they should still stay.

Reply:
I meant the first parameter being passed to the constructor. There are two implementations of filestore now, so the type can be set in the classes themselves. 

Otherwise you can create an ephemeral one and have its files marked as permanent in the backing store.